src/ProjectTemplates/BlazorTemplates.Tests/BlazorWasmTemplateTest.cs

```diff
@@ -10,43 +10,43 @@
 using System.Net.Http.Headers;
 using System.Text.Json;
 using System.Text.RegularExpressions;
-using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.E2ETesting;
+using Microsoft.AspNetCore.BrowserTesting;
 using Microsoft.AspNetCore.Internal;
 using Microsoft.AspNetCore.Testing;
 using Microsoft.Extensions.CommandLineUtils;
 using Newtonsoft.Json.Linq;
-using OpenQA.Selenium;
-using OpenQA.Selenium.Support.Extensions;
+using PlaywrightSharp;
+using ProjectTemplates.Tests.Infrastructure;
 using Templates.Test.Helpers;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Templates.Test
 {
-    public class BlazorWasmTemplateTest : BrowserTestBase
+    public class BlazorWasmTemplateTest
     {
-        public BlazorWasmTemplateTest(ProjectFactoryFixture projectFactory, BrowserFixture browserFixture, ITestOutputHelper output)
-            : base(browserFixture, output)
+        public BlazorWasmTemplateTest(ProjectFactoryFixture projectFactory, PlaywrightFixture<BlazorServerTemplateTest> browserFixture, ITestOutputHelper output)
         {
             ProjectFactory = projectFactory;
+            Fixture = browserFixture;
+            Output = output;
+            BrowserContextInfo = new ContextInformation(TestHelpers.CreateFactory(output));
         }
 
         public ProjectFactoryFixture ProjectFactory { get; set; }
+        public PlaywrightFixture<BlazorServerTemplateTest> Fixture { get; }
+        public ITestOutputHelper Output { get; }
+        public ContextInformation BrowserContextInfo { get; }
 
-        public override Task InitializeAsync()
-        {
-            return InitializeAsync(isolationContext: Guid.NewGuid().ToString());
```

Comment:
Could you clarify - is there an equivalent isolation mechanism in the new Playwright infrastructure for ensuring we're not inheriting any cached data from a preceding test run? That's necessary for the PWA caching test to be valid.

Reply:
BrowserContexts are "incognito-like" sessions, so they are isolated from each other by default and are much cheaper to create/destroy than an entire browser.